app/views/subscriptions/new.html.erb

```diff
@@ -0,0 +1,58 @@
+<h1>Add Billing info</h1>
+<%= form_tag subscription_path, method: :post, id: "payment-form" do %>
+  <span class="payment-errors"></span>
+  <div class="form-row">
+    <label>
+      <span>Card Number</span>
+      <input id="number" type="text" size="20" data-stripe="number"/>
+    </label>
+  </div>
+
+  <div class="form-row">
+    <label>
+      <span>CVC</span>
+      <input id="cvc" type="text" size="4" data-stripe="cvc"/>
+    </label>
+  </div>
+
+  <div class="form-row">
+    <label>
+      <span>Expiration (MM/YYYY)</span>
+      <input id="exp-month" type="text" size="2" data-stripe="exp-month"/>
+    </label>
+    <span> / </span>
+    <input id="exp-year" type="text" size="4" data-stripe="exp-year"/>
+  </div>
+
+  <button type="submit">Save Card</button>
+<% end %>
+
+<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
+<script type="text/javascript">
+  Stripe.setPublishableKey('<%= Rails.configuration.stripe[:publishable_key] %>');
+  $(function($) {
+    $('#payment-form').submit(function(event) {
+      var $form = $(this);
+      // Disable the submit button to prevent repeated clicks
+      $form.find('button').prop('disabled', true);
+      Stripe.card.createToken($form, stripeResponseHandler);
+      // Prevent the form from submitting with the default action
+      return false;
+    });
+  });
+  function stripeResponseHandler(status, response) {
+    var $form = $('#payment-form');
+    if (response.error) {
+      // Show the errors on the form
+      $form.find('.payment-errors').text(response.error.message);
+      $form.find('button').prop('disabled', false);
+    } else {
+      // response contains id and card, which contains additional card details
+      var token = response.id;
+      // Insert the token into the form so it gets submitted to the server
+      $form.append($('<input type="hidden" name="stripeToken" />').val(token));
+      // and submit
+      $form.get(0).submit();
+    }
+  };
+</script>
```

Comment:
also an end of line here, think we could make this a form_for with a subscription model.
And i do like forms in a DRY way, only when it gets to hard to keep them in the same place it's good to split them up